static/js/publisher/release/reducers/pendingReleases.js

```diff
@@ -2,85 +2,149 @@ import {
   RELEASE_REVISION,
   UNDO_RELEASE,
   CANCEL_PENDING_RELEASES,
-  SET_PROGRESSIVE_RELEASE_PERCENTAGE
+  SET_PROGRESSIVE_RELEASE_PERCENTAGE,
+  UPDATE_PROGRESSIVE_RELEASE_PERCENTAGE,
+  PAUSE_PROGRESSIVE_RELEASE,
+  RESUME_PROGRESSIVE_RELEASE
 } from "../actions/pendingReleases";
+
 import { CLOSE_CHANNEL } from "../actions/pendingCloses";
 
 function removePendingRelease(state, revision, channel) {
-  if (state[revision.revision]) {
-    const channels = [...state[revision.revision].channels];
-
-    if (channels.includes(channel)) {
-      state = { ...state };
-      channels.splice(channels.indexOf(channel), 1);
-      state[revision.revision] = {
-        ...state[revision.revision],
-        channels
-      };
+  const newState = { ...state };
```

Comment:
`{ ...state }` only makes a shallow copy of the state, so any modifications on child object will be leaked.
Better to use `JSON.parse(JSON.stringify(state))` hack. We probably could extract it to some `deepClone` utility. 
```suggestion
  const newState = JSON.parse(JSON.stringify(state));
```